fdbserver/storageserver.actor.cpp

```diff
@@ -4175,6 +4185,329 @@ Key constructMappedKey(KeyValueRef* keyValue,
 	return mappedKeyTuple.pack();
 }
 
+ACTOR Future<Void> validateRangeAgainstServer(StorageServer* data,
+                                              KeyRange range,
+                                              Version version,
+                                              StorageServerInterface remoteServer) {
+	TraceEvent(SevDebug, "ValidateRangeAgainstServerBegin", data->thisServerID)
+	    .detail("Range", range)
+	    .detail("Version", version)
+	    .detail("RemoteServer", remoteServer.toString());
+
+	state int validatedKeys = 0;
+	state std::string error;
+	loop {
+		try {
+			state int limit = 1e4;
+			state int limitBytes = CLIENT_KNOBS->REPLY_BYTE_LIMIT;
+			state GetKeyValuesRequest req;
+			req.begin = firstGreaterOrEqual(range.begin);
+			req.end = firstGreaterOrEqual(range.end);
+			req.limit = limit;
+			req.limitBytes = limitBytes;
+			req.version = version;
+			req.tags = TagSet();
+			state Future<ErrorOr<GetKeyValuesReply>> remoteKeyValueFuture =
+			    remoteServer.getKeyValues.getReplyUnlessFailedFor(req, 2, 0);
+			state ErrorOr<GetKeyValuesReply> remoteResult = wait(remoteKeyValueFuture);
+
+			state GetKeyValuesRequest localReq;
+			localReq.begin = firstGreaterOrEqual(range.begin);
+			localReq.end = firstGreaterOrEqual(range.end);
+			localReq.limit = limit;
+			localReq.limitBytes = limitBytes;
+			localReq.version = version;
+			localReq.tags = TagSet();
+			data->actors.add(getKeyValuesQ(data, localReq));
+			GetKeyValuesReply local = wait(localReq.reply.getFuture());
```

Comment:
Probably it's better to wait `localReq` and `remoteKeyValueFuture` together than in sequence.

Reply:
Change applied.